### PR TITLE
Fix duplication

### PR DESCRIPTION
**Pull request by KaoruToda**
add_attribute_object and add_DN_object have similar code.
I put similar code in build_data().

### CODE REVIEW

File: apps/req.c [hunk 1]
```diff
@@ -1153,56 +1158,22 @@ static int add_DN_object(X509_NAME *n, char *text, const char *def,
...
-    buf[--i] = '\0';
-#ifdef CHARSET_EBCDIC
-    ebcdic2ascii(buf, buf, i);
-#endif
-    if (!req_check_len(i, n_min, n_max)) {
-        if (batch || value)
-            return 0;
-        goto start;
-    }
+    const char *desc1 = "DN value";
```
Comment (paulidale) [inline apps/req.c]: Why aren't these inline in the function call?  There is no need to have local variables for the descriptions.

File: apps/req.c [hunk 2]
```diff
@@ -1153,56 +1158,22 @@ static int add_DN_object(X509_NAME *n, char *text, const char *def,
...
+    const char *desc2 = "DN default";
+
+    ret = build_data(text, def, value, n_min, n_max, buf, sizeof(buf), desc1, desc2);
+    if (ret == 0)
```
Comment (paulidale) [inline apps/req.c]: How about:

```
if (ret == 0 || ret == 1)
    return ret;
```

the same thing occurs later...

File: apps/req.c [hunk 3]
```diff
@@ -62,6 +62,11 @@ static int add_DN_object(X509_NAME *n, char *text, const char *def,
                          char *value, int nid, int n_min, int n_max,
                          unsigned long chtype, int mval);
 static int genpkey_cb(EVP_PKEY_CTX *ctx);
+static int build_data(char *text, const char *def,
+                         char *value, int n_min, int n_max,
```
Comment (paulidale) [inline apps/req.c]: indentation?

**Comment by paulidale** [comment]
Less code is better :)

File: apps/req.c (hunk 1 above)
Comment (KaoruToda) [reply apps/req.c]: @paulidale Thank you for review.

These variables of the description are used in join() when buffer size is small.
Error message are 4 patterns.
- add_attribute_object has 2 patterns.
- add_DN_object has 2 patterns.

Comment (paulidale) [reply apps/req.c]: I don't think I worded it well.  I didn't mean to inline them into build_data itself, just in the call to build_data:
```
ret = build_data(text, def, value, n_min, n_max, buf, sizeof(buf), "DN value", "DN default");
```

and similarly for the other case.

They are only used once as an argument for the build_data call, this doesn't warrant a local variable IMO.

Comment (KaoruToda) [reply apps/req.c]: @paulidale Thank you for the explanation. I think so, too. And fix it.

**Comment by levitte** [comment]
This needs a rebase.  I'm prepared to approve after that's done.

And apologies for the long wait, @KaoruToda

**Comment by KaoruToda** [comment]
@levitte I have finished to rebase.
However CI is failed.
Do I have to resolve conflicts?

**Comment by levitte** [comment]
Yes, you have to resolve conflicts.  And then push the update (you will have to use the `-f` flag when doing so).

**Comment by KaoruToda** [comment]
@levitte I have finised to resolve conflicts.

**Comment by mspncp** [comment]
Hello @KaoruToda,

in the OpenSSL repository merge commits like 466daea744e6 are prohibited. You have to rebase your branch to the tip of master instead. Fortunately, in git almost every mistake can be easily undone: Here are the necessary steps for you to revert the merge commit, rebase your branch on top of the master branch, and force-push your changes to your OpenSSL GitHub repository.

**update your local copy and the master branch**

```
git fetch --all
git checkout master
git pull
```

**checkout your pull request branch**

```
git checkout fix-duplication
```

**verify that the tip of your branch points to the merge commit**

```
git show

    commit 466daea744e60b28c06f6250bfcf3306bbff6bea (HEAD -> fix-duplication, github/pr-4566)
    Merge: 9a4a43f958 5848be0488
    Author: Kaoru Toda <kunnpuu@gmail.com>
    Date:   Thu Mar 29 20:50:20 2018 +0900

        Merge branch 'master' into fix-duplication
```

**undo the merge commit**

```
git reset --hard HEAD~
```


**rebase your branch on top of the master branch**

```
git rebase master
```

**force-push your new branch to your github clone**

```
git push origin fix-duplication --force
```

**Comment by KaoruToda** [comment]
＠mspncp Thank you for the advice. Is my pushing correct?

**Comment by mspncp** [comment]
Yes, everything is fine now. GitHub now shows [only one commit](https://github.com/openssl/openssl/pull/4566/commits) for your pull request and what is most important is the green check mark saying "This branch has no conflicts with the base branch". 

ping @levitte for approval

**Comment by KaoruToda** [comment]
@mspncp I was relieved to hear that. Thank you so much.

**Comment by richsalz** [comment]
I didn't realize that Ben was the other reviewer, so I'll wait for @levitte or someone else to also review.

**Comment by richsalz** [comment]
Thank you for helping improve the code!  Merged to master.

**Pull request closed without merging**